code/modules/atmospherics/machinery/air_alarm/_air_alarm.dm

```diff
@@ -131,6 +131,25 @@ GLOBAL_LIST_EMPTY_TYPED(air_alarms, /obj/machinery/airalarm)
 	GLOB.air_alarms += src
 	check_enviroment()
 
+/obj/machinery/airalarm/add_context_self(datum/screentip_context/context, mob/user)
+	if(isnull(context.held_item))
+		return
```

Comment:
Instead of checking if the player has an item in the context's themselves, we should put the behaviour in the context object to standardise it. I'm thinking having something to track if players have tools to only show the tips if they have a tool on them

Reply:
Yeah. I feel like having all the tools be on screen at once is too much visual noise.

If you can come up with something so I don't have to istype every two lines for a new tool, that'd be useful